pilot/pkg/model/authorization.go

```diff
@@ -86,22 +90,21 @@ func (policy *AuthorizationPolicies) ListAuthorizationPolicies(selectionOpts Wor
 		return configs
 	}
 
-	rootNamespace := policy.RootNamespace
-	wlNamespace := selectionOpts.WorkloadNamespace
-	svcNamespace := selectionOpts.ServiceNamespace
-	var lookupInNamespaces []string
-
-	if svcNamespace != "" {
-		lookupInNamespaces = []string{svcNamespace}
-	} else if wlNamespace != rootNamespace {
-		// Only check the root namespace if the (workload) namespace is not already the root namespace
-		// to avoid double inclusion.
-		lookupInNamespaces = []string{rootNamespace, wlNamespace}
-	} else {
-		lookupInNamespaces = []string{wlNamespace}
+	if len(selectionOpts.Services) > 1 {
+		// Currently, listing multiple services is unnecessary.
+		// To simplify, this function allows at most one service.
+		// The restriction can be lifted if future needs arise.
+		panic("ListAuthorizationPolicies expects at most 1 service in WorkloadPolicyMatcher")
+	}
+
+	lookupInNamespaces := sets.New[string]()
```

Comment:
Same comment as above around non-deterministic -- here, I think it does matter though. I don't see where we sort the config anywhere after this function

Reply:
I see you have the sortAPByNamespaceAndName but its only used for tests which illustrates the issue.

I am thinking instead of  a set we still want a list, just one with no duplicates. We could just do add a `slices.FilterDuplicates()` maybe and use that (we have a `FilterDuplicatesPresorted` but it won't work since we are not sorted)?

Reply:
I’ve fixed this as suggested.
Thanks for you feedback.  :-)